File: backend/domain/src/main/kotlin/pt/ira/documents/Documents.kt
package pt.ira.documents

/**
 * Representa um documento de apoio.
 *
 * Um documento é um recurso informativo disponibilizado aos averiguadores,
 * organizado por categorias (como automóvel, saúde ou legislação de referência).
 * Armazena os metadados necessários para a sua catalogação, localização e acesso.
 *
 * @property id Identificador único do documento.
 * @property name Nome descritivo do documento (ex: "Declaração de Acidente", "Código da Estrada").
 * @property type Categoria do documento (ex: "Automovel", "Pessoal", "Legislacao").
 * @property filepath Caminho ou localização do ficheiro no sistema de armazenamento.
 *
 * @constructor Cria uma instância de [Documents] com os metadados necessários
 *              para servir o documento aos averiguadores.
 */

data class Documents(
    val id: Int,
    val name: String,
    val type: String,
    val filepath: String,
)

File: frontend/commons/models/documents/Documents.ts
/**
 * Representa um documento armazenado no sistema.
 * @property id Identificador único do documento.
 * @property name Nome do documento.
 * @property type Tipo do documento.
 * @property filepath Caminho do ficheiro no servidor.
 */
export interface Documents {
  id: number;
  name: string;
  type: string;
  filepath: string;
}
